.github/ISSUE_TEMPLATE/bug_report.md

```diff
@@ -10,23 +10,66 @@ assignees: ''
 
 The release number or commit hash of the version you're using.
 
-## Operating system & version
+## \[Optional\] Setup information
 
-Ex: Windows 10, Ubuntu 18.04, iOS 14.
+If you are reporting a bug about:
+
+- difficulties getting Bevy to build or run on your machine
+- unusual rendering bugs
+- unusual input bugs
+- hardware specific problems
+
+These issues are often specific to hardware or OS configuration and can be challenging to reproduce.
+As a result, please consider including information about:
+
+- the Rust version you're using
+  - Bevy has a policy of relying on the "latest stable release" of Rust
+  - Nightly should generally work, but there are sometimes regressions: please let us know!
+- the operating system, including its version
+  - e.g. Windows 10, Ubuntu 18.04, iOS 14.
+- relevant hardware
+  - if your bug is rendering-related, copy the adapter info that appears when you run Bevy
+  - e.g. `AdapterInfo { name: "NVIDIA GeForce RTX 2070", vendor: 4318, device: 7938, device_type: DiscreteGpu, backend: Vulkan }` or XBox series X controller
+
+You should also consider testing the examples of our upstream dependencies to help isolate the issue:
+
+- [`wgpu`](https://github.com/gfx-rs/wgpu) for rendering problems
+- [`winit`](https://github.com/rust-windowing/winit) for input and window management
+- [`gilrs`](https://docs.rs/gilrs/latest/gilrs/) for gamepad inputs
 
 ## What you did
 
-The steps you took to uncover this bug. Please list full reproduction steps if
-feasible.
+The steps you took to uncover this bug.
+Please provide a runnable snippet that demonstrates the problem if you can.
 
-## What you expected to happen
+For example:
 
-What you think should've happened if everything was working properly.
+```rust
+use bevy::prelude::-;
 
-## What actually happened
+fn main(){
+    App::new()
+    .add_plugins(DefaultPlugins)
+    .run();
+}
+```
 
-The actual result of the actions you described.
+If you can't produce a minimal reproduction, linking to a repository can also be very helpful.
```

Comment:
I reworded the initial request to include linking to an example.

Reply:
```suggestion
```